optical_flow.rs

```diff
@@ -0,0 +1,127 @@
+use crate::spi::HypedSpi;
+use std::time::{Duration, Instant, SystemTime, UNIX_EPOCH};
+use byteorder::{LittleEndian, ReadBytesExt};
+use std::thread::sleep;
+use serde::{Serialize, Deserialize};
+use serde_json::json;
+
+
+// implement get_motion function first
+// add readByteFromRegister (?) function to HypedSPI trait which takes register address and gives data
+
+const REG_MOTION_BURST: u8 = 0x16;
+const TIMEOUT: Duration = Duration::from_secs(5);
+
+
+pub struct PMW3901<SPI> {
```

Comment:
```suggestion
pub struct PMW3901<SPI: HypedSpi> {
```
It's also standard to have single uppercase characters as type parameters, like `T`. But I don't mind `SPI` for now

Reply:
Probs better to call this `OpticalFlow` also